tests/test_packages/test_skills/test_coin_price/test_behaviours.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the tests of the behaviour classes of the coin price skill."""
+
+from pathlib import Path
+from typing import cast
+
+from aea.test_tools.test_skill import BaseSkillTestCase
+
+from packages.fetchai.protocols.http.message import HttpMessage
+from packages.fetchai.protocols.prometheus.message import PrometheusMessage
+from packages.fetchai.skills.coin_price.behaviours import CoinPriceBehaviour
+
+from tests.conftest import ROOT_DIR
+
+
+class TestSkillBehaviour(BaseSkillTestCase):
+    """Test behaviours of coin price."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "coin_price")
+
+    @classmethod
+    def setup(cls, **kwargs):
+        """Setup the test class."""
+        super().setup()
+        cls.coin_price_behaviour = cast(
+            CoinPriceBehaviour, cls._skill.skill_context.behaviours.coin_price_behaviour
+        )
+
+    def test_send_http_request_message(self):
+        """Test the send_http_request_message method of the coin_price behaviour."""
+        self.coin_price_behaviour.send_http_request_message("GET", "some_url")
+        self.assert_quantity_in_outbox(1)
+        msg = cast(HttpMessage, self.get_message_from_outbox())
+        assert msg, "Wrong message type"
+        assert (
+            msg.performative == HttpMessage.Performative.REQUEST
+        ), "Wrong message performative"
+        assert msg.url == "some_url", "Wrong url"
```

Comment:
Suggestion, you can generally combine all the attributes you are expecting from your message via `message_has_attributes()`, then whatever attribute you want checked you provide to this method as arguments. E.g.
```
has_attributes, error_str = self.message_has_attributes(
            actual_message=msg,
            message_type= HttpMessage,
            performative= HttpMessage.Performative.REQUEST,
            to=...,
            sender=...,
            url=""some_url"",
        )
        assert has_attributes, error_str
```